File: apps/backend/src/strategy/Strategy.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { RebalanceFrequency, StrategyType, UUID } from '@mid/shared';

@Entity()
export class Strategy {
  @PrimaryGeneratedColumn('uuid')
  id: UUID;

  @Column({ type: 'varchar', length: 200 })
  name: string;

  @Column({ type: 'varchar', length: 1000 })
  description?: string;

  @Column({ type: 'varchar', length: 200 })
  rebalanceFrequency: RebalanceFrequency;

  @Column({ type: 'varchar', length: 200 })
  type: StrategyType;

  @Column({ type: 'text', nullable: true })
  algorithm?: string;

  @CreateDateColumn({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  createdAt: Date;
}

File: backend/src/main/kotlin/com/kyoulho/mid/strategy/Strategy.kt
package com.kyoulho.mid.strategy

import jakarta.persistence.*
import java.util.*

@Entity
data class Strategy(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: String? = null,

    @Column(length = 200, nullable = false)
    val name: String,

    @Column(length = 1000)
    val description: String,

    @Column(length = 200, nullable = false)
    val rebalanceFrequency: RebalanceFrequency,

    @Column(length = 200, nullable = false)
    @Enumerated(EnumType.STRING)
    val type: StrategyType,
)
